foundation-ui/mount/MountService.js

```diff
@@ -0,0 +1,123 @@
+import {EventEmitter} from 'events';
+import {CHANGE} from './MountEvent';
+
+/**
+ * MountService
+ */
+class MountService extends EventEmitter {
+  constructor() {
+    super(...arguments);
+
+    /**
+     * Private Context
+     *
+     * @typedef {Object} MountService~Context
+     * @property {MountService} instance - Current mount service instance
+     * @property {Array.<{component:function, priority:Number, type:String}>}
+     *     components -  List of component descriptors
+     * @property {number} rank - Consecutive number used to rank components
+     *     based on order of registration
+     */
+    const context = {
+      components: [],
+      instance: this,
+      rank: 0
+    };
+
+    this.registerComponent = this.registerComponent.bind(context);
+    this.unregisterComponent = this.unregisterComponent.bind(context);
+    this.findComponentsWithType = this.findComponentsWithType.bind(context);
+  }
+
+  /**
+   * Register component with type and optional priority
+   *
+   * @this MountService~Context
+   * @param {function} component - React component which will be
+   * mounted based on the provided type
+   * @param {String} type - String which is used to match components with the
+   * respective mounts
+   * @param {Number} [priority] - Number which is used to sort the components
+   */
+  registerComponent(component, type, priority = 0) {
```

Comment:
I like that we're going to have homogeneous services, same code style, same public API style 👍